dashboard/app/controllers/scripts_controller.rb

```diff
@@ -236,13 +233,17 @@ def get_unit
   def set_unit
     @script = get_unit
     raise ActiveRecord::RecordNotFound unless @script
+  end
 
-    if current_user && @script.pilot? && !@script.has_pilot_access?(current_user)
-      render :no_access
-    end
+  def render_no_access
+    if current_user
+      if  @script.pilot? && !@script.has_pilot_access?(current_user)
+        return render :no_access
+      end
 
-    if current_user && @script.in_development? && !current_user.permission?(UserPermission::LEVELBUILDER)
-      render :no_access
+      if @script.in_development? && !current_user.permission?(UserPermission::LEVELBUILDER)
+        return render :no_access
+      end
```

Comment:
again, it looks like this logic belongs in can can, at least for the routes besides `show`

Reply:
Why would show want a different message than everything else? Just because thats keeping things behaving the same?